pkg/controller/istio/helper.go

```diff
@@ -3,27 +3,52 @@ package istio
 import (
 	"crypto/sha256"
 	"encoding/hex"
-	"github.com/operator-framework/operator-sdk/pkg/k8sutil"
-	"k8s.io/apimachinery/pkg/runtime/schema"
 	"os"
 	"strconv"
 	"strings"
+
+	"github.com/operator-framework/operator-sdk/pkg/k8sutil"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/rest"
 )
 
 var (
+	// VirtualServiceGVK => definition of virtual service GVK for oneagent
 	VirtualServiceGVK = schema.GroupVersionKind{
 		Group:   "networking.istio.io",
 		Version: "v1alpha3",
 		Kind:    "VirtualService",
 	}
 
+	// ServiceEntryGVK => definition of virtual service GVK for oneagent
 	ServiceEntryGVK = schema.GroupVersionKind{
 		Group:   "networking.istio.io",
 		Version: "v1alpha3",
 		Kind:    "ServiceEntry",
 	}
+
+	istioGVRName = "networking.istio.io"
 )
 
+// CheckIstioEnabled - Checks if Istio is installed
+func CheckIstioEnabled(cfg *rest.Config) (bool, error) {
+
+	client := discovery.NewDiscoveryClientForConfigOrDie(cfg)
```

Comment:
Don't kill the process because of this. `discovery.NewDiscoveryClientForConfig()` works here. Just return the error if it fails.